src/app/iteration/iteration.service.ts

```diff
@@ -80,10 +81,35 @@ export class IterationService {
   }
 
   /**
-   * Get locally saved iterations' reference
+   * Update an existing iteration
+   * @param iteration - Updated iteration
+   * @return updated iteration's reference from the list
    */
-  getIterationList() {
-    return this.iterations;
+  updateIteration(iteration: IterationModel): Promise<IterationModel> {
+    return this.http
+      .patch(iteration.links.self, { headers: this.headers })
+      .toPromise()
+      .then (response => {
+        if (/^[5, 4][0-9]/.test(response.status.toString())) {
```

Comment:
As I explained earlier, this check and error is only for testing. The mock http error response is not leading the promise to catch section. I can't figure out why it's not happening. So, if it's a live error from API and it has a meaningful message it goes to catch section by default and it'll never come to this check.
What do you mean by "regex is not for free" ?